Workaround
src/hscurl.c:50:11: error:
     error: expected ';' after top level declarator
      "darcs/" PACKAGE_VERSION " libcurl/" LIBCURL_VERSION;
              ^
              ;
   |
50 |   "darcs/" PACKAGE_VERSION " libcurl/" LIBCURL_VERSION;
   |           ^
1 error generated.
`cc' failed in phase `C Compiler'. (Exit code: 1)

This seems related to a long closed http://bugs.darcs.net/issue1824
titled "PACKAGE_VERSION is not a string"

I don't see any value in reporting the version of darcs to the server
anyway, so implement the current effective behavior of darcs on Windows.

Index: darcs.cabal
--- darcs.cabal.orig
+++ darcs.cabal
@@ -467,7 +467,7 @@ Library
       other-modules:        Darcs.Util.Download.Curl
       cpp-options:          -DHAVE_CURL
       c-sources:            src/hscurl.c
-      cc-options:           -DHAVE_CURL
+      cc-options:           -DHAVE_CURL -DPACKAGE_VERSION="unknown"
       if flag(pkgconfig)
         pkgconfig-depends:  libcurl
       else
